### ISSUE TITLE
Posting a event that extends AbstractEvent ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by clienthax** [SpongePowered/SpongeForge#389]
Seems to be because the event class has no interfaces and the method assumes that it does
```
            EventManager eventManager = Sponge.getGame().getEventManager();
            eventManager.post(new VotifierEvent(fakeVote));
```

test event
``` java
package com.vexsoftware.votifier.model;

import org.spongepowered.api.event.impl.AbstractEvent;

/**
 * {@code VotifierEvent} is a custom Sponges event class that is sent
 * synchronously to Sponges's main thread allowing other plugins to listener
 * for votes.
 * 
 * @author frelling
 * 
 */
public class VotifierEvent extends AbstractEvent {
	/**
	 * Encapsulated vote record.
	 */
	private Vote vote;

	/**
	 * Constructs a vote event that encapsulated the given vote record.
	 * 
	 * @param vote
	 *            vote record
	 */
	public VotifierEvent(final Vote vote) {
		this.vote = vote;
	}

	/**
	 * Return the encapsulated vote record.
	 * 
	 * @return vote record
	 */
	public Vote getVote() {
		return vote;
	}

}
```

```
[17:30:27] [Server thread/ERROR] [Sponge]: Error occurred while executing command 'fakevote clienthax' for source EntityPlayerMP['clienthax'/321, l='world', x=-30.04, y=72.00, z=-5.75]: 0
java.lang.ArrayIndexOutOfBoundsException: 0
	at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:232) ~[SpongeModEventManager.class:?]
	at com.swifteh.GAL.Commands.onCommand(Commands.java:230) ~[Commands.class:?]
	at com.swifteh.GAL.Commands.process(Commands.java:38) ~[Commands.class:?]
	at org.spongepowered.api.util.command.dispatcher.SimpleDispatcher.process(SimpleDispatcher.java:340) ~[SimpleDispatcher.class:?]
	at org.spongepowered.api.service.command.SimpleCommandService.process(SimpleCommandService.java:250) [SimpleCommandService.class:?]
	at net.minecraft.command.ServerCommandManager.executeCommand(ServerCommandManager.java:80) [ServerCommandManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.handleSlashCommand(NetHandlerPlayServer.java:853) [NetHandlerPlayServer.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processChatMessage(NetHandlerPlayServer.java:829) [NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.C01PacketChatMessage.processPacket(C01PacketChatMessage.java:44) [C01PacketChatMessage.class:?]
	at net.minecraft.network.play.client.C01PacketChatMessage.processPacket(C01PacketChatMessage.java:57) [C01PacketChatMessage.class:?]
	at net.minecraft.network.PacketThreadUtil$1.onProcessPacket(PacketThreadUtil.java:53) [PacketThreadUtil$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:24) [PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_60]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_60]
	at net.minecraftforge.fml.common.FMLCommonHandler.callFuture(FMLCommonHandler.java:714) [FMLCommonHandler.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:727) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:385) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:669) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:540) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_60]
[17:30:27] [Netty Server IO #3/ERROR] [FML]: NetworkDispatcher exception
```

**Comment by ST-DDT** [comment]
Could you please append the stacktrace as well?

**Comment by JBYoshi** [comment]
Should be fixed in #390.
